backend/lib/plugins/auth.js

```diff
@@ -30,16 +36,24 @@ const authPlugin = async (app) => {
     }),
   });
 
+  app.decorateRequest("userId", null);
+
   app.decorate("authenticate", async (req, reply) => {
     try {
-      await req.jwtVerify();
-      const { user } = req;
-      req.userId = mongoose.Types.ObjectId(user[MONGO_ID_KEY]);
+      await checkAuth(req);
     } catch (err) {
       reply.send(err);
     }
   });
 
+  app.decorate("authenticateOptional", async (req) => {
+    try {
+      await checkAuth(req);
+    } catch (err) {
+      req.userId = null;
```

Comment:
In the optional auth, if there is no user authenticated, the `req.userId` field will be absent or null?

Reply:
it will be `null` by default and it will also be `null`if this hook cannot verify the JWT and decode it.